src/openms/source/ANALYSIS/ID/FragmentIndexScorer.cpp

```diff
@@ -284,13 +284,25 @@ namespace OpenMS
   }
 */
 
-  void FragmentIndexScorer::trimHits(OpenMS::FragmentIndexScorer::SpectrumMatchesTopN& init_hits)
+  void FragmentIndexScorer::trimHits(OpenMS::FragmentIndexScorer::SpectrumMatchesTopN& init_hits) const
   {
-    buildKMinHeap<SpectrumMatch, uint32_t>(init_hits.hits_, max_processed_hits_, [](SpectrumMatch a) { return a.num_matched_; });
     if (max_processed_hits_ < init_hits.hits_.size())
     {
+      std::partial_sort(init_hits.hits_.begin(), init_hits.hits_.begin() + max_processed_hits_, init_hits.hits_.end(), [](SpectrumMatch a, SpectrumMatch b){
+        return a.num_matched_ > b.num_matched_;
```

Comment:
this likely leads to non-deterministic sorting on different platforms in case num_matched peaks are identical.
To make it deterministic one usually includes more fields in the comparison.
e.g., return std::tie(a.num_matched, a.otherfield ..., -1.0 * a.anotherfields ...) > std::tie(b.num_matched, b.otherfield ..., -1.0 * b.anotherfields ...)